packages/cli-kit/src/public/node/environments.ts

```diff
@@ -111,3 +112,43 @@ export async function getEnvironmentNames(fileName: string, options?: LoadEnviro
 
   return Object.keys(environments)
 }
+
+export const ENVIRONMENT_ALLOWED_GLOB_PATTERN = /[*?[\]{}]/
+
+/**
+ * Expands environment patterns (including globs) to actual environment names.
+ * @param patterns - Array of environment names or glob patterns.
+ * @param fileName - The file name to load environments from.
+ * @param options - Optional configuration for loading.
+ * @returns Array of matched environment names.
+ */
+export async function expandEnvironmentPatterns(
+  patterns: string[],
+  fileName: string,
+  options?: LoadEnvironmentOptions,
+): Promise<string[]> {
+  const allEnvironments = await getEnvironmentNames(fileName, options)
+  const matchedEnvironments = new Set<string>()
+
+  for (const pattern of patterns) {
+    const isGlob = ENVIRONMENT_ALLOWED_GLOB_PATTERN.test(pattern)
```

Comment:
Instead of matching against this we could *always* parse with `minimatch` but this would result in us ignoring the flag if it doesn't find any matches.

While this is open for review I'm going to play around with the consequences of that approach.